### PR TITLE
Fix rustdoc stabilized versions layout

### LINKED ISSUE #86342: rustdoc: stabilized (added) versions of enum variants are hard to read
Go to: https://doc.rust-lang.org/nightly/std/io/enum.ErrorKind.html#variant.OutOfMemory
It was [stabilized (i.e. added) in 1.54.0](https://github.com/rust-lang/rust/blob/607d6b00d4e0e0475b8de9d0c870b7126fdcdf6b/library/std/src/io/error.rs#L190-L193) but the rustdoc generates like:

the added versions of enum variants are hard to read

The problem is that the previous variant's version is placed at the same height. I think it's ideal to place the version of the item itself at the same height, like functions.

### PR DESCRIPTION
**Pull request by matteo-briani**
Fixes #86342
r? @GuillaumeGomez

### CODE REVIEW

File: src/librustdoc/html/static/rustdoc.css [hunk 1]
```diff
@@ -902,7 +907,6 @@ body.blur > :not(#help) {
 	font-size: initial;
 	position: absolute;
 	right: 0;
-	top: 0;
```
Comment (GuillaumeGomez) [inline src/librustdoc/html/static/rustdoc.css:905]: Instead of removing this, please add `position: relative;` to `.small-section-header {`.

Comment (matteo-briani) [reply src/librustdoc/html/static/rustdoc.css:905]: Sure, thanks for the hint

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Waiting for CI to pass then approving it.

File: src/librustdoc/html/static/rustdoc.css (hunk 1 above)
Comment (matteo-briani) [reply src/librustdoc/html/static/rustdoc.css]: I mistakenly force-pushed, it should have been an extra commit as I read from @rust-highfive  ...my bad!

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/rustdoc.css]: Not an issue. :)

**Pull request merged**